File: scripts/r/analisis_bibliometrico_mapas_tematicos_y_acoplamiento.R
library(bibliometrix)
library(wordcloud2)

biblioshiny()

M <- mergeDbSources(scopus, wos, remove.duplicated = TRUE)

s <- biblioAnalysis(M, sep = ";")
summary(s)

res <- thematicMap(M, field = "ID", n = 10, minfreq = 10, size = 0.5, repel = TRUE)
plot(res$net$graph_pajek)

R <- authorProdOverTime(M)

write.csv2(M, "df_bibliometrico.csv")

couplingMap(M, analysis = "documents", field = "CR", n = 100,
            impact.measure = "local", minfreq = 10, size = 0.5, repel = TRUE)

wordcloud2(df, size = 0.45)

File: scripts/python/analisis_bibliometrico_mapas_tematicos_y_acoplamiento.py
import pandas as pd
import matplotlib.pyplot as plt
from wordcloud import WordCloud
import bibliometrics as bm

M = bm.merge_sources([bm.load("scopus.csv"), bm.load("wos.csv")], remove_duplicated=True)

s = bm.biblio_analysis(M, sep=";")
print(s.summary())

res = bm.thematic_map(M, field="ID", n=10, minfreq=10, size=0.5, repel=True)
res["graph"].plot()

R = bm.author_prod_over_time(M)

M.to_csv("df_bibliometrico.csv", sep=";")

bm.coupling_map(M, analysis="documents", field="CR", n=100,
                impact_measure="local", minfreq=10, size=0.5, repel=True)

df = pd.read_csv("df.csv")
wc = WordCloud(width=1000, height=700).generate_from_frequencies(dict(zip(df["term"], df["freq"])))

plt.figure(figsize=(10, 7))
plt.imshow(wc, interpolation="bilinear")
plt.axis("off")
plt.show()
